src/app/modules/submission/email-submission/__tests__/email-submission.routes.spec.ts

```diff
@@ -0,0 +1,368 @@
+  createTransport: jest.fn().mockReturnValue({
+    sendMail: jest.fn(),
+  }),
+}))
+
+// TODO (#149): Import router instead of creating it here
+const SUBMISSIONS_ENDPT_BASE = '/v2/submissions/email'
+const SUBMISSIONS_ENDPT = `${SUBMISSIONS_ENDPT_BASE}/:formId([a-fA-F0-9]{24})`
+
+const MOCK_SUBMISSION_BODY = {
+  responses: [],
+  isPreview: false,
+}
+
+const EmailSubmissionsRouter = Router()
+EmailSubmissionsRouter.post(
+  SUBMISSIONS_ENDPT,
+  CaptchaFactory.validateCaptchaParams,
+  FormController.formById,
+  PublicFormMiddleware.isFormPublicCheck,
+  CaptchaMiddleware.checkCaptchaResponse as RequestHandler,
+  SpcpController.isSpcpAuthenticated,
+  EmailSubmissionsMiddleware.receiveEmailSubmission,
+  celebrate({
+    body: Joi.object({
+      responses: Joi.array()
+        .items(
+          Joi.object()
+            .keys({
+              _id: Joi.string().required(),
+              question: Joi.string().required(),
+              fieldType: Joi.string()
+                .required()
+                .valid(...Object.values(BasicField)),
+              answer: Joi.string().allow(''),
+              answerArray: Joi.array(),
+              filename: Joi.string(),
+              content: Joi.binary(),
+              isHeader: Joi.boolean(),
+              myInfo: Joi.object(),
+              signature: Joi.string().allow(''),
+            })
+            .xor('answer', 'answerArray') // only answer or answerArray can be present at once
+            .with('filename', 'content'), // if filename is present, content must be present
+        )
+        .required(),
+      isPreview: Joi.boolean().required(),
+    }),
+  }),
+  EmailSubmissionsMiddleware.validateEmailSubmission,
+  MyInfoController.verifyMyInfoVals as RequestHandler,
+  (SubmissionsController.injectAutoReplyInfo as unknown) as RequestHandler,
+  SpcpController.appendVerifiedSPCPResponses as RequestHandler,
+  EmailSubmissionsMiddleware.prepareEmailSubmission as RequestHandler,
+  EmailSubmissionsMiddleware.saveMetadataToDb,
+  EmailSubmissionsMiddleware.sendAdminEmail,
+  SubmissionsController.sendAutoReply,
+)
+
+const EmailSubmissionsApp = setupApp('/', EmailSubmissionsRouter)
+
+describe('email-submission.routes', () => {
+  let request: Session
+
+  beforeAll(async () => await dbHandler.connect())
+  beforeEach(() => {
+    request = session(EmailSubmissionsApp)
+  })
+  afterEach(async () => {
+    await dbHandler.clearDatabase()
+    jest.restoreAllMocks()
+  })
+  afterAll(async () => await dbHandler.closeDatabase())
+
+  describe('SPCP authentication', () => {
+    describe('SingPass', () => {
+      it('should return 200 when submission is valid', async () => {
+        mockSpClient.verifyJWT.mockImplementationOnce((_jwt, cb) =>
+          cb(null, {
+            userName: 'S1234567A',
+          }),
+        )
+        const { form } = await dbHandler.insertEmailForm()
+        // authType cannot be saved if form is published, so need to save twice
```

Comment:
Not sure if you want to do this in this PR (or ever), but we can overload the `insertEmailForm` method to accept all parameters of Form.create haha then can override with your own meta before saving:

```ts
const insertEmailForm = async ({
  ...,
  otherProps = {},
}: {
  ...,
  otherProps?: Partial<IEmailFormSchema>
} = {}): Promise<{...}> => {
  ...
  const form = await EmailFormModel.create({
    title: '...',
    ...,
    ...otherProps, // <---
  })
}
```

Reply:
true, I'll do that